packages/react-instantsearch-hooks-server/src/getServerState.tsx

```diff
@@ -182,22 +182,21 @@ function getInitialResults(rootIndex: IndexWidget): InitialResults {
 }
 
 function importRenderToString() {
-  return Promise.all([
-    // React pre-18 doesn't use `exports` in package.json, requiring a fully resolved path
-    // Thus, only one of these imports is correct
-    // eslint-disable-next-line import/extensions
-    import('react-dom/server.js').catch(() => {}),
-    import('react-dom/server').catch(() => {}),
-  ]).then((imports) => {
-    const ReactDOMServer = imports.find(
-      (mod): mod is { renderToString: typeof RenderToString } =>
-        mod !== undefined
-    );
+  // React pre-18 doesn't use `exports` in package.json, requiring a fully resolved path
+  // Thus, only one of these imports is correct
+  const modules = ['react-dom/server.js', 'react-dom/server'];
```

Comment:
Can we also mention why we need this intermediary variable to avoid regressions?